docs/adr/ADR-002_Scan_repository_for_hardcoded_secrets.md

```diff
@@ -0,0 +1,148 @@
+# ADR-002: Scan repository for hardcoded secrets
+
+>|              | |
+>| ------------ | --- |
+>| Date         | `05/06/2023` |
+>| Status       | `RFC` |
+>| Deciders     | `Engineering` |
+>| Significance | `Construction techniques` |
+>| Owners       | `Dan Stefaniuk, Jon Pearce, Tamara Goldschmidt, Tim Rickwood` |
+
+---
+
+- [ADR-002: Scan repository for hardcoded secrets](#adr-002-scan-repository-for-hardcoded-secrets)
+  - [Context](#context)
+  - [Decision](#decision)
+    - [Assumptions](#assumptions)
+    - [Drivers](#drivers)
+    - [Options](#options)
+    - [Outcome](#outcome)
+    - [Rationale](#rationale)
+  - [Consequences](#consequences)
+  - [Compliance](#compliance)
+  - [Notes](#notes)
+  - [Actions](#actions)
+  - [Tags](#tags)
+
+## Context
+
+To safeguard sensitive details like passwords, API keys etc. from being incorporated into code repositories, it is imperative that we employ secret scanning of the code. This safeguarding process should be conducted in two key areas. Firstly, on the developer's machine, we utilise a git pre-commit hook to halt the inclusion of any secrets within the committed code. Secondly, as a safety net, a similar scan should be integrated into the CI/CD pipeline. Should a secret be detected within this pipeline, it is crucial that the pipeline serves as a gate to fail the build, subsequently blocking any related pull requests.
+
+## Decision
+
+### Assumptions
+
+There is already a well-known and fit-for-purpose tool `git-secrets` in use that was selected as the outcome of a decision made around 4 years ago. The purpose of this document is to review that decision.
+
+### Drivers
+
+Within NHS England, we are observing an adoption of the `gitleaks` tool, which is an alternative to `git-secrets`.
+
+### Options
+
+There are three options presented in this decision record.
+
+1. [git-secrets](https://github.com/awslabs/git-secrets)
+
+   - Repository metadata
+     - Contributions
+       - Number of contributors: **28**
+       - Number of commits: **110**
+       - Commit dates / frequency: **last commit more than a half a year ago, very low frequency**
+       - Number of Stars & Forks: **11.1k & 1.1k**
+     - Implementation technologies: **Shell script**
+     - Licence: **[Apache-2.0](https://choosealicense.com/licenses/apache-2.0/)**
+   - Features
+     - [x] Scan whole history
+     - [x] Scan single commit
+     - [ ] Predefined set of rules: _A very limited number of rules_
+     - [x] Definition of custom rules
+     - [x] Definition of custom exclusions patterns
+     - [ ] Entropy detection
+     - [ ] Pre-backed Docker image
+
+   - Pros
+     - A well know tool that has been around for a while
+   - Cons
+     - Rules and exclusion patterns are not easy to manage as no comments or metadata are allowed in the definition
+     - No pre-backed Docker image
+
```

Comment:
Maybe add:
- Activity of the repo has dropped (last commit a while ago)

Reply:
Added. Thanks